src/main/java/org/spongepowered/api/map/MapView.java

```diff
@@ -0,0 +1,331 @@
+     * <p>The coordinates start with 0,0 as the top left corner moving right
+     * and down.</p>
+     *
+     * @param x The x coordinate of the pixel
+     * @param y The y coordinate of the pixel
+     * @param color The color to be matched before setting the pixel to the closest {@link MapColor}
+     * @throws IndexOutOfBoundsException If the coordinates are greater than
+     *         the width and height of the map's canvas
+     */
+    void setPixelSimilar(int x, int y, Color color);
+
+    /**
+     * Returns the {@link MapColor} at the specified pixel. An unset pixel will be returned as
+     * {@link MapColors#AIR}
+     *
+     * The coordinates start with 0,0 as the top left corner moving right and
+     * down.
+     *
+     * @param position The coordinates of the pixel to get
+     * @return The color of the pixel at the specified coordinates
+     * @throws IndexOutOfBoundsException If the coordinates are greater than or equal to
+     * the width and height of the map's canvas
+     */
+    default MapColor getPixel(Vector2i position) {
+        return getPixel(position.getX(), position.getY());
+    }
+
+    /**
+     * Returns the {@link MapColor} at the specified pixel. An unset pixel will be returned as
+     * {@link MapColors#AIR}
+     *
+     * The coordinates start with 0,0 as the top left corner moving right and
+     * down.
+     *
+     * @param x The x coordinate of the pixel to get
+     * @param y The y coordinate of the pixel to get
+     * @return The color of the pixel at the specified coordinates
+     * @throws IndexOutOfBoundsException If the coordinates are greater than or equal to
+     * the width and height of the map's canvas
+     */
+    MapColor getPixel(int x, int y);
+
+    /**
+     * Create an {@link BufferedImage} out of the map allowing it to be viewed outside
+     * the game in conventional image viewers and shared on the web. Simply
+     * converts indexes in the map's canvas to the RGB values they represent.
+     *
+     * @return The rendered image
+     */
+    BufferedImage toImage();
+
+    /**
+     * Draws an image to the map, starting at the top left corner,
+     * clipping it if it goes out of bounds.
+     *
+     * @param image The image to draw
+     */
+    default void drawImage(BufferedImage image) {
+        drawImage(0, 0, image);
+    }
+
+    /**
+     * Draws an image to the map, clipping it if it goes out of bounds.
+     *
+     * @param position The top left coordinate on the map
+     * @param image The image to draw
+     */
+    default void drawImage(Vector2i position, BufferedImage image) {
+        drawImage(position.getX(), position.getY(), image);
+    }
+
+    /**
+     * Draws an image to the map, clipping it if it goes out of bounds.
+     *
+     * @param x The top left x coordinate on the map
+     * @param y The top left y coordinate on the map
+     * @param image The image to draw
+     */
+    void drawImage(int x, int y, BufferedImage image);
```

Comment:
This raises some more complicated questions, because of how BufferedImage works, the only way to get alpha is to access the alpha on the raster itself. In other words, I have to specifically handle every pixel format (the order of the channels), and pixel type (the way the channels are packed into the buffer).

While I plan to handle the most common types in this way for speed (should save a copy), the fallback is to copy the pixels out using `BufferedImage#toRGB` which doesn't specifically mention if the RGB pixel is actually an ARGB pixel, but seems to imply it.

Whatever happens with implementation of it, how about a method taking a threshold for alpha to do a 1bit alpha masking, along with this default? The existing method will ignore alpha and blit all pixels, since that can't be replicated with a threshold.